### DISCUSSION TITLE
iOS 构建失败

### DISCUSSION THREAD

**Discussion by ChainHerman** [alibaba/xquic#442] [Q&A]
编译报错 CMake Error at third_party/boringssl/CMakeLists.txt:776 (install):
  install TARGETS given no BUNDLE DESTINATION for MACOSX_BUNDLE executable
  target "bssl".

我的所有命令如下

heqian@heqiandeMacBook-Pro ~ % cd /Users/heqian/Desktop/MG/Quic 
heqian@heqiandeMacBook-Pro Quic % git clone https://github.com/alibaba/xquic.git 
heqian@heqiandeMacBook-Pro Quic % cd xquic
heqian@heqiandeMacBook-Pro xquic % git clone https://github.com/google/boringssl.git ./third_party/boringssl
heqian@heqiandeMacBook-Pro xquic % cd ./third_party/boringssl
heqian@heqiandeMacBook-Pro boringssl % mkdir -p build && cd build
heqian@heqiandeMacBook-Pro build % cmake -DBUILD_SHARED_LIBS=0 -DCMAKE_C_FLAGS="-fPIC" -DCMAKE_CXX_FLAGS="-fPIC" ..
heqian@heqiandeMacBook-Pro build % make ssl crypto
heqian@heqiandeMacBook-Pro build % cd ..
heqian@heqiandeMacBook-Pro boringssl % SSL_TYPE_STR="boringssl"
heqian@heqiandeMacBook-Pro boringssl % SSL_PATH_STR="${PWD}"
heqian@heqiandeMacBook-Pro boringssl % cd ../..
heqian@heqiandeMacBook-Pro xquic % sh xqc_build.sh ios build artifacts $(pwd)/third_party/boringssl
heqian@heqiandeMacBook-Pro xquic % Brew install cmake       
heqian@heqiandeMacBook-Pro xquic % export IOS_CMAKE_TOOLCHAIN=/Users/heqian/Desktop/MG/Quic/xquic/cmake/ios.toolchain.cmake
heqian@heqiandeMacBook-Pro xquic % sh xqc_build.sh ios build artifacts $(pwd)/third_party/boringssl

这时候报错 CMake Error at third_party/boringssl/CMakeLists.txt:776 (install):
  install TARGETS given no BUNDLE DESTINATION for MACOSX_BUNDLE executable
  target "bssl".

然后我就修改了 cmakeLists 的776 行为  install(TARGETS bssl
    RUNTIME DESTINATION ${INSTALL_DESTINATION_DEFAULT}/bin
    LIBRARY DESTINATION ${INSTALL_DESTINATION_DEFAULT}/lib
    BUNDLE DESTINATION ${INSTALL_DESTINATION_DEFAULT}/bin)

然后再次构建
heqian@heqiandeMacBook-Pro xquic % sh xqc_build.sh ios ios_build ios_release /Users/heqian/Desktop/MG/Quic/xquic/third_party/boringssl 

再次报错
In file included from /Users/heqian/Desktop/MG/Quic/xquic/src/transport/xqc_engine.h:12:
In file included from /Users/heqian/Desktop/MG/Quic/xquic/src/tls/xqc_tls.h:9:
In file included from /Users/heqian/Desktop/MG/Quic/xquic/third_party/boringssl/include/openssl/err.h:114:
/Users/heqian/Desktop/MG/Quic/xquic/third_party/boringssl/include/openssl/base.h:81:10: fatal error: 'boringssl_prefix_symbols.h' file not found
#include <boringssl_prefix_symbols.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/heqian/Desktop/MG/Quic/xquic/src/http3/xqc_h3_conn.c:5:
In file included from /Users/heqian/Desktop/MG/Quic/xquic/src/http3/xqc_h3_conn.h:11:
In file included from /Users/heqian/Desktop/MG/Quic/xquic/src/transport/xqc_conn.h:8:
In file included from /Users/heqian/Desktop/MG/Quic/xquic/third_party/boringssl/include/openssl/ssl.h:145:
/Users/heqian/Desktop/MG/Quic/xquic/third_party/boringssl/include/openssl/base.h:81:10: fatal error: 'boringssl_prefix_symbols.h' file not found
#include <boringssl_prefix_symbols.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
1 error generated.
make[2]: *** [CMakeFiles/xquic.dir/src/http3/xqc_h3_stream.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/xquic.dir/src/http3/xqc_h3_conn.c.o] Error 1
make[1]: *** [CMakeFiles/xquic.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

试了许多次都是这样，求助

**Comment by lurker-Chen** [comment]
xqc_build.sh里使用了BORINGSSL_PREFIX的能力，有两种方式：
1、手动修改下xqc_build.sh，删掉-DBORINGSSL_PREFIX和-DBORINGSSL_PREFIX_SYMBOLS两行；
2、在编译boringssl的时候增加下编译参数-DBORINGSSL_PREFIX=tb -DBORINGSSL_PREFIX_SYMBOLS=boringssl_sympols.txt
